portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,234 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+// PROJECT SLIDESHOW LOGIC
+let slideIndex = 1;
+showSlides(slideIndex);
+
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+function currentSlide(n) {
+  showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+  let i;
+  let slides = document.getElementsByClassName("project");
+  let dots = document.getElementsByClassName("dot");
+
+  if (n > slides.length) {
+      slideIndex = 1;
+  }
+  if (n < 1) {
+      slideIndex = slides.length;
+  }
```

Comment:
I think it's better semantically if checking whether n is out of range is handled in changeSlide/plusSlide. showSlide is a function that sounds like it should just handle the display, whereas changeSlide function handles these edge cases logic. Having this kind of side effects in showSlide may be confusing in the future.

Reply:
Ah yes good point here! Will make the change 👍